soc/xtensa/intel_adsp/cavs/gdbstub.c

```diff
@@ -0,0 +1,1394 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <xtensa-asm2-context.h>
+
+static struct xtensa_register gdb_reg_list[] = {
+/* XT-GDB has different register definitions than Zephyr toolchain GDB */
+#ifdef CONFIG_ADSP_XT_GDB
+	{
+		/* PC */
+		.idx = 32,
+		.regno = 0x0020,
+		.byte_size = 4,
+		.gpkt_offset = 0,
+		.stack_offset = BSA_PC_OFF,
+	},
+	{
+		/* AR0 */
+		.idx = 0x100,
```

Comment:
Style: not understanding the mixing of decimal and hex literals for the index and regno fields.  Would be good to just pick one.